File: python/src/stack.py
from typing import List


def execute(program: List[str]) -> List[int]:
    # initialize the stack
    stack = [1, 2, 3, 4, 5, 6, 7]
    for instruction in program:
        if instruction == "peek":
            # print out the top item in stack
            if stack:
                print(stack[-1])
            else:
                print("Stack is empty!")
        elif instruction == "pop":
            # pop the top item in stack
            if stack:
                stack.pop()
            else:
                print("Cannot pop from an empty stack!")
        else:
            if instruction.startswith("push"):
                try:
                    data = int(instruction[5:])
                    stack.append(data)
                except ValueError:
                    print(f"Invalid data format: {instruction[5:]}")
            else:
                print(f"Invalid instruction: {instruction}")

    return stack


if __name__ == "__main__":
    try:
        num_instructions = int(input("Enter the number of instructions: "))
        program = [input(f"Instruction {i+1}: ") for i in range(num_instructions)]
        res = execute(program)
        print(" ".join(map(str, res)))
    except ValueError:
        print("Invalid number of instructions!")

File: typescript/src/stack.ts
function Stack(params: string[]): number[] {
	let stack: number[] = [];
	for (let instruction of params) {
		if (instruction == "peek") {
			if (stack.length > 0) {
				console.log(stack[stack.length - 1]);

			} else {
				console.log("stack is empty");

			}

		} else if (instruction == "pop") {
			if (stack.length > 0) {
				stack.pop();
			} else {
				console.log("cannot pop");

			}

		} else {
			if (instruction.startsWith("push")) {
				const dataString: string = instruction.slice(5);
				const data: number = parseFloat(dataString);
				if (!isNaN(data)) {
					stack.push(data);
				} else {
					console.log("invalid data format");

				}
				console.log("invalid string");


			}

		}
	}
	return stack;
}
